src/compiler/checker.ts

```diff
@@ -12863,7 +12862,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
         else {
             mapper = createTypeMapper(typeParameters, typeArguments);
-            members = createInstantiatedSymbolTable(source.declaredProperties, mapper, /*mappingThisOnly*/ typeParameters.length === 1);
+            const propertiesMapper = isTupleType(type)
+                ? createTypeMapper(typeParameters, sameMap(typeArguments, (t, i) => addOptionality(t, /*isProperty*/ true, !!(type.target.elementFlags[i] & ElementFlags.Optional))))
+                : mapper;

```

Comment:
To make the overall change **smaller** in scope we could "flip" the situation here. The existing `mapper` could be used as `propertiesMapper` (without the current changes in this PR it would have the `missingType` in the type arguments) and we could `removeMissingType` from the "core" `mapper` here. This way `T` in `Tuple<T>` wouldn't contain the `missingType` but properties would still be instantiated with it.

Alternatively, it's also very likely that we could add the `missingType` to the `tuple.target` elements at optional elements. This way we could just have a single `mapper` here and the `missingType` would be added to tuple properties at the instantiation time of the `.target`'s fixed elements. Currently, target is represented as smth like `[?, ?, ?]` - where `?` are type parameters. It could be represented as `[?, ?, ? | missingType]` though. I'm still battling myself how to think about it - would it be a cool trick or an obfuscated solution? 😅 The trick is that the instantiated tuple still wouldn't have `missingType` in its type arguments but the properties would be automatically read from the instantiated `.target` - so there would be a tricky mismatch between the tuple's type arguments and its `.target`. I'm not even sure if that's even possible - I didn't actually try to do it this way, I distinctly recall having such an idea 2 days ago when analyzing this.

That said, I still very much like that the `missingType` gets removed from the type arguments of the tuple. It feels right to me. The optionality bit is added by the element flag - and it's not part of the "core" type at that position. This way, It would feel a lot closer to me to how it works with object properties where the optionality gets added [here](https://github.dev/microsoft/TypeScript/blob/5d04803196899fab535103a5114bc9bf8d04dbd9/src/compiler/checker.ts#L10795-L10797). It's part of the computed property type - but not part of the `declaredType` there.